File: dart/2966.dart
class Solution {
    List<List<int>> divideArray(List<int> n, int k) {
        int l = n.length;
        n.sort();
        List<List<int>> r = [];
        for (int i = 0, j = 0; i < l; i += 3, j++) {
            if (n[i + 2] - n[i] > k)
                return [];
            r.add([ n[i], n[i + 1], n[i + 2] ]);
        }
        return r;
    }
}

File: python3/2966.py
class Solution:
    def divideArray(self, n: List[int], k: int) -> List[List[int]]:
        l = len(n)
        n.sort()
        b = int(l / 3)
        for i in range(0, l, 3):
            if n[i + 2] - n[i] > k:
                return []
        return [[n[i], n[i + 1], n[i + 2]] for i in range(0, l, 3)]
